core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -2097,3 +2101,39 @@ unfoldMany (Unfold ustep inject) (Fold fstep initial extract) =
 
     {-# INLINE_LATE consume #-}
     consume s a = inject a >>= produce s
+
+-- | Fold the input stream to top n elements.
+--
+-- >>> stream = Stream.fromList [2::Int,7,9,3,1,5,6,11,17]
+-- >>> Stream.fold (Fold.top 3) stream >>= MA.toList
+-- [17,11,9]
+--
+-- /Pre-release/
+top :: (MonadIO m, Storable a, Ord a) => Int -> Fold m a (MA.Array a)
+top n = Fold step initial extract
```

Comment:
Add a `topBy cmp n` and then define `top = topBy compare`. If we want to find out bottom n elements we can use `topBy (flip compare)`.